sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -857,15 +887,7 @@ public Object call(SQLDatabase db) throws Exception {
         } catch (InterruptedException e) {
             throw new RuntimeException(e);
         } catch (ExecutionException e) {
-            if(e.getCause() != null){
-                if(e.getCause() instanceof DocumentNotFoundException){
-                    throw (DocumentNotFoundException)e.getCause();
-                } else {
-                    throw new RuntimeException(e);
-                }
-            } else {
-                throw new RuntimeException(e);
-            }
+           throw new DocumentNotFoundException(String.format("Could not find document with id: %s",docId),e);
```

Comment:
If this is the only exception it could be, why not just rethrow the one from inside the callable?